rfd/xxxx-bot-for-cert-renewals.md

```diff
@@ -0,0 +1,633 @@
+| Private key (TLS)      | `key`        |                                  |
+
+The bot then generates a set of non-renewable credentials for each configured
+credential destination. In the initial implementation, the following artifacts
+may or may not be generated depending on the user's configuration (kind and
+templates).
+
+| Kind                   | Path          | Notes                                   | Kind | Config Template
+| -----------------------|---------------|-----------------------------------------|------|---------
+| Private key            | `key`         | Shared for both SSH and TLS uses        | Both | Always
+| SSH public key         | `key.pub`     | Required for OpenSSH compatibility      | SSH  | Always
+| SSH client certificate | `sshcert`     | Signed by Teleport `UserCA`             | SSH  | Always
+| SSH known hosts        | `known_hosts` | Teleport `HostCA` certs, OpenSSH format | SSH  | `ssh-client`
+| SSH client config      | `ssh_config`  | `Include`-ready SSH client config       | SSH  | `ssh-client`
+| TLS CA certificates    | `tlscacerts`  | Teleport CA certificates (both)         | TLS  | Always
+| TLS client certificate | `tlscert`     | Signed by Teleport `UserCA`             | TLS  | Always
+
+In the case of `directory` destinations, we encourage the use of POSIX ACLs.
+This is meant to ensure that the primary (renewable) credentials are
+exclusively readable to the bot while the secondary (non-renewable)
+credentials are readable only to the target user but still writable by the
+bot. The `tbot init` subcommand can be used to configure ACLs automatically
+where supported.
+
+Notes:
+ * Users may optionally decide to disable persistence of the renewable
+   certificate using the `memory` storage backend. This has security benefits
+   but prevents the bot from recovering if stopped as the renewable credentials
+   would only exist in memory.
+    * Future joining methods (e.g. EC2 identity document) may mitigate this
+      downside.
+ * The Teleport auth server's CA certificates are verified at first connect
+   using CA pins, following Teleport's usual node joining procedure.
+ * Unneeded certificate types/formats could be optionally disabled in
+   `tbot.yaml` config if desired.
+ * Additional artifacts may be written if users enable any app-specific config
+   templates.
+ * In the future, we could consider ephemeral certificate destinations for
+   situations where certificates should not be written to disk (Webhooks or
+   other IPC?)
+
+#### Reload Events
+
+To enforce security best practices, the bot itself provides no builtin methods
+for signalling or otherwise executing user code when a certificate rotation
+occurs. Instead, end users are encouraged to use filesystem watches (e.g.
+`inotify`) to watch for changes directly using a least-privileged system user.
+We will include a helper command, `tbot watch`, to perform simple actions on
```

Comment:
Will the command be just part of a “standalone” utility? What are the privileges that the launched script/command will have?

Reply:
`tbot watch` will be effectively standalone and will just provide a nice UX on top of `inotify`. It'll inherit privileges of the Unix user that runs it, but otherwise does nothing you couldn't also do with other tools, e.g. [`inotifywatch`](https://linux.die.net/man/1/inotifywatch).